app/controllers/sessions_controller.rb

```diff
@@ -2,6 +2,9 @@
 
 class SessionsController < ApplicationController
   before_action :check_error_result, :require_code_param, :exchange_token, only: [:result]
+  before_action :check_session_expiration, except: [:timeout]
```

Comment:
I was wondering how the inheritance works with the same before_action in ApplicationController and the subcontroller. The Rails guide suggests to use `skip_before_action` here:
```suggestion
  skip_before_action :check_session_expiration, only: [:timeout]
```